tests/bluetooth/controller/uut/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# CMakeLists.txt file for creating of uut library.
+#
+
+add_library(uut STATIC
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/util/mem.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/util/memq.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_tx_queue.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_conn.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ll_feat.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_pdu.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_local.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_phy.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_common.c
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/ull_llcp_chmu.c
+)
+
+target_include_directories(uut PUBLIC
+  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/include
+  ${ZEPHYR_BASE}/tests/bluetooth/controller/common/include
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/include
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller
+  ${ZEPHYR_BASE}/subsys/bluetooth
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw
+  ${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/nordic
+  ${ZEPHYR_BASE}/include/zephyr/bluetooth
+)
+
+add_subdirectory(${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl mocks)
```

Comment:
Comment for future reference.

This is not ideal, as we should strive to have a straight forward CMake hierarchy.
But as this is an improvement on the overall test structure, then it is accepted in this case.

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/53741 for followup